File: tests/fixtures/python_hello/app/greeter.py
"""Greeter implementation – used by the SpecSlice Python indexer fixture."""

from dataclasses import dataclass

from .utils import banner


@dataclass
class GreeterConfig:
    name: str
    formal: bool = False


class Greeter:
    """A toy class with a couple of methods so callHierarchy has something
    to chew on when a real Python LSP server is available."""

    def __init__(self, config: GreeterConfig) -> None:
        self.config = config

    def greet(self) -> str:
        prefix = banner(self.config.formal)
        return f"{prefix}, {self.config.name}!"

    def goodbye(self) -> str:
        return f"Bye, {self.config.name}!"


def make_greeter(name: str, *, formal: bool = False) -> Greeter:
    return Greeter(GreeterConfig(name=name, formal=formal))


async def make_greeter_async(name: str) -> Greeter:
    return make_greeter(name)

File: tests/fixtures/typescript_hello/src/greeter.ts
import { banner } from "./utils";

export interface GreeterConfig {
  readonly name: string;
  readonly formal?: boolean;
}

export class Greeter {
  constructor(private readonly config: GreeterConfig) {}

  greet(): string {
    const prefix = banner(this.config.formal ?? false);
    return `${prefix}, ${this.config.name}!`;
  }

  goodbye(): string {
    return `Bye, ${this.config.name}!`;
  }
}

export function makeGreeter(name: string, formal = false): Greeter {
  return new Greeter({ name, formal });
}

export async function makeGreeterAsync(name: string): Promise<Greeter> {
  return makeGreeter(name);
}
